arch/risc-v/src/c906/c906_start.c

```diff
@@ -73,6 +73,10 @@ void __c906_start(uint32_t mhartid)
   const uint32_t *src;
   uint32_t *dest;
 
+  /* Configure FPU */
+
+  riscv_fpuconfig();
```

Comment:
For C906, I'll update this patch to remove relative code in head.S

For MPFS, it not only enabled the FPU but also set XS bit, so it's better to keep origin implementation.

Reply:
But should we remove FPU related config from MPFS/head.S too to unify the FPU initialization process?

Reply:
OK, FPU relative flags setting removed from mpfs_head.S.

Reply:
@pussuw could you look the change in mpfs is good?

Reply:
This does change the functionality of our "bootloader" app, which is compiled without FPU, but still enables it for harts 1,2,3,4. The new functionality in this PR if probably more correct in that respect (not enable the FPU if built without the FPU support), but I am not able to test it this week if it breaks something since I am travelling without HW. If @pussuw is ok with it, it is fine with me as well.

Reply:
I also do not have a very simple way to test this change but at least the differences I noticed is that 1. hart0 in mpfs does not have an fpu and we handle that in mpfs_head.S. The other change is that we mark the FPU as dirty, but init should be fine too.